docs/architecture.md

```diff
@@ -0,0 +1,54 @@
+# Architecture
+
+The language features provided by the [MicroProfile Language Server](./microprofile.ls) depend on the dependencies in the user's Java Project.
+For example, MicroProfile Fault Tolerance features (snippets, validation, completion, etc.) are only provided if the user's
+Java project has MicroProfile Fault Tolerance as a dependency.
+
+In order to avoid duplicating the work done by an IDEs Java Language Support (computing the classpath, rebuilding AST, etc.) the MicroProfile language Server relies on an external component that understands and handles LSP command requests sent from the MicroProfile Language Server.
+
+The MicroProfile Language Server has limited knowledge of the users Java project, and instead uses the external component for information on the Java project. In the context of Eclipse and VS Code, this external component is managed with [MicroProfile JDT](./microprofile.jdt).
+The external component does not necessarily require the JDT however, as different implementations can be used instead. For example [Quarkus Tools for IntelliJ](https://github.com/redhat-developer/intellij-quarkus) does not use the JDT and instead has its own implementation used to handle the requests from [`MicroProfileLanguageClientAPI`](https://github.com/eclipse/lsp4mp/blob/master/microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/api/MicroProfileLanguageClientAPI.java). In other words, the MicroProfile Language Server **does not** depend on JDT.
+
+## microprofile-config.properties language features
+
+For properties completion, the MicroProfile Language Server uses the external component to detect all properties available on the classpath.
+
```

Comment:
I will add 

The external component compute and return those properties by following the same format than [Spring Boot Configuration Metadata](https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-configuration-metadata.html#configuration-metadata-format).

```json
{
	"properties": [
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.model.reader",
			"description": "Configuration property to specify the fully qualified name of the OASModelReader implementation."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.filter",
			"description": "Configuration property to specify the fully qualified name of the OASFilter implementation."
		},
		{
			"type": "boolean",
			"defaultValue": "false",
			"extensionName": "microprofile-openapi-api",
			"required": false,
			"name": "mp.openapi.scan.disable",
			"description": "Configuration property to disable annotation scanning. Default value is `false`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.scan.packages",
			"description": "Configuration property to specify the list of packages to scan.\r\nFor example, `mp.openapi.scan.packages=com.xyz.PackageA,com.xyz.PackageB`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.scan.classes",
			"description": "Configuration property to specify the list of classes to scan.\r\nFor example, `mp.openapi.scan.classes=com.xyz.MyClassA,com.xyz.MyClassB`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.scan.exclude.packages",
			"description": "Configuration property to specify the list of packages to exclude from scans.\r\nFor example, `mp.openapi.scan.exclude.packages=com.xyz.PackageC,com.xyz.PackageD`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.scan.exclude.classes",
			"description": "Configuration property to specify the list of classes to exclude from scans.\r\nFor example, `mp.openapi.scan.exclude.classes=com.xyz.MyClassC,com.xyz.MyClassD`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.servers",
			"description": "Configuration property to specify the list of global servers that provide connectivity information.\r\nFor example, `mp.openapi.servers=https://xyz.com/v1,https://abc.com/v1`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.servers.path.{*}",
			"description": "Prefix of the configuration property to specify an alternative list of servers to service all operations in a path.For example, `mp.openapi.servers.path./airlines/bookings/{id}=https://xyz.io/v1`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.servers.operation.{*}",
			"description": "Prefix of the configuration property to specify an alternative list of servers to service an operation.Operations that want to specify an alternative list of servers must define an `operationId`, a unique string used to identify the operation.For example, `mp.openapi.servers.operation.getBooking=https://abc.io/v1`."
		},
		{
			"type": "java.lang.String",
			"extensionName": "microprofile-openapi-api",
			"required": true,
			"name": "mp.openapi.schema.{*}",
			"description": "Prefix of the configuration property to specify a schema for a specific class, in JSON format.The remainder of the property key must be the fully-qualified class name.The value must be a valid OpenAPI schema object, specified in the JSON format.The use of this property is functionally equivalent to the use of the `@Schema` annotationon a Java class, but may be used in cases where the application developer does not have access to the source code of a class."
		}
	]
}
```

Reply:
Perhaps in an another PR, but I think it should be nice to have a simple diagram with vscode completion which shows MP completion and LSP4MP server side which returns Spring Boot Metadata JSON list of properties.